File: yunu/프로그래머스/다음에 올 숫자/내 코드.js
function solution(common) {
  const [f1, f2, e2, e1] = [common[0], common[1], common.at(-2), common.at(-1)];
  if (f1 + e1 === f2 + e2) {
    return common.at(-1) + (f2 - f1);
  }
  return common.at(-1) * (f2 / f1);
}

File: yunu/프로그래머스/숫자 변환하기/내 코드.py
from collections import deque

def solution(x, y, n):
    cache = [-1 for _ in range(y + 1)]
    cache[x] = 0
    
    queue = deque([x])
    while queue:
        num1 = queue.popleft()
        if num1 == y: return cache[y]
        for num2 in [num1 + n, num1 * 2, num1 * 3]:
            if num2 > y: continue
            if cache[num2] != -1: continue
            cache[num2] = cache[num1] + 1
            queue.append(num2)
    
    return -1
    